libp2p/mplex/channel.nim

```diff
@@ -0,0 +1,58 @@
+## Nim-LibP2P
+## Copyright (c) 2018 Status Research & Development GmbH
+## Licensed under either of
+##  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+##  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+## at your option.
+## This file may not be copied, modified, or distributed except according to
+## those terms.
+
+import chronos
+import ../connection, ../stream
+import mplex
+
+type
+  Channel = ref object of LPStream
```

Comment:
This is a bit outdated now and I haven't pushed the new version because there are too many things in flight. But the idea is to use the buffered stream as the underlying abstraction for multiplexed connections. Tho the channel is mplex specific it would extend buffered stream.